Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs

```diff
@@ -70,10 +74,38 @@ import Tools.Whatsapp
 data AuthReq = AuthReq
   { mobileNumber :: Text,
     mobileCountryCode :: Text,
-    merchantId :: ShortId Merchant --,
+    merchantId :: ShortId Merchant,
+    deviceToken :: Maybe FCMRecipientToken,
+    whatsappNotificationEnroll :: Maybe Whatsapp.OptApiMethods,
+    firstName :: Maybe Text,
+    middleName :: Maybe Text,
+    lastName :: Maybe Text,
+    email :: Maybe Text,
+    language :: Maybe Maps.Language,
+    gender :: Maybe SP.Gender
     -- otpChannel :: Maybe OTPChannel
   }
-  deriving (Generic, FromJSON, ToJSON, Show, ToSchema)
+  deriving (Generic, ToJSON, Show, ToSchema)
+
+instance A.FromJSON AuthReq where
+  parseJSON v = case v of
+    A.Object obj ->
+      AuthReq <$> obj .: "mobileNumber"
+        <*> obj .: "mobileCountryCode"
+        <*> obj .: "merchantId"
+        <*> obj .:? "deviceToken"
+        <*> obj .:? "whatsappNotificationEnroll"
+        <*> obj .:? "firstName"
+        <*> obj .:? "middleName"
+        <*> obj .:? "lastName"
+        <*> obj .:? "email"
+        <*> obj .:? "language"
+        <*> obj .:? "gender"
+    A.String s ->
```

Comment:
Why?

Reply:
Basically Req can come in Object or String and it has to be then decoded in AuthReq so made this FromJSON such that It is able to decode both the types of Request Body (String as well as JSON). This is done so that business logic can remain same in case of signature authentication, just verification flow can be added.

Type 1 : JSON
{ "mobileNumber": "9642429378", "mobileCountryCode": "+91", "merchantId" : "NAMMA_YATRI" }

Type 2 : Stringified JSON
"{\"mobileNumber\":\"9642429378\",\"mobileCountryCode\":\"+91\",\"merchantId\":\"NAMMA_YATRI\",\"timestamp\":\"2023-04-13T07:28:40+00:00\"}"